docs/content/en/frontend/storefront-api.md

```diff
@@ -27,435 +27,159 @@ php artisan vendor:publish --tag="shopify-theme"
-
-```js
-const handleProductFormSubmit = form => {
-  const quantity = form.querySelector('#ss-product-qty')
-  const variantId = form.querySelector('#ss-product-variant')
-
-  if (variantId == null) {
-    return
-  }
-
-  const lineItemsToAdd = [
-    {
-      variantId: variantId.value,
-      quantity: quantity != null ? parseInt(quantity.value) : 1
-    }
-  ]
-
-  client.checkout
-    .addLineItems(checkoutId, lineItemsToAdd)
-    .then(checkout => {
-      const elements = htmlToElements(
-        '<p><span class="mr-2">Product added to the basket.</span><a href="/cart">Go to cart</a></p>'
-      )
-      bannerMessage(elements, true)
-      setCartCount(checkout.lineItems)
-    })
-    .catch(err => {
-      // Handle Errors here.
-    })
-}
-```
-
-<alert scope="info">
-
-We use two functions `bannerMessage()` and `setCartCount()` which we will address further down the page. In short, these display feedback to the user.
-
-</alert>
-
-## Cart.js
+## cart.js
 
-This is the big one. We handle pulling the cart details into our site, as well as updating quantity, managing deletion of products and handing off to Shopify.
+This file provides methods for creating, updating and reading from the cart associated with the current user.
 
-### setCartCount
+You can import any of the following methods. Note that all methods are `async` so returns promises.
 
-In the demo theme, we add a count into the header so the user can see how many products are in the basket. This function returns the cart, checks the total quantity and sets the count to that value.
+`addLines(string: cartId, array: lines)`
+This method lets you add an array of lines to the cart. See [Shopify's documentation](https://shopify.dev/docs/api/storefront/2024-01/mutations/cartLinesAdd) for more information.
 
-```js
-const setCartCount = () => {
-  const countTarget = document.querySelector('[data-ss-cart-count]')
-
-  if (countTarget == null) {
-    return
-  }
-
-  client.checkout
-    .fetch(checkoutId)
-    .then(({ lineItems }) => {
-      let count = 0
-      lineItems.forEach(item => (count = count + item.quantity))
-      countTarget.innerHTML = count
-    })
-    .catch(err => {
-      // Handle Errors here.
-    })
-}
-```
+`createFreshCart(?array: lines)`
+This method will make a totally new cart. If an array of cart lines are passed then they will be added to the cart.
 
-### hideCartOverview
+`getExistingCart(string: cartId)`
+This returns gets the cart identified by the passed id.
 
-Toggles whether the cart should be displayed. This is used upon loading the cart page.
+`getOrCreateCart(?string: cartId)`
+This method will get the cart that matches the id, or if there is none or no id is passed then  an fresh cart will be returned.
+   
+`removeLine(string cartId, string lineId)`
+This method will return the line identified by line id from the cart.
```

Comment:
@ryanmitchell This doesn't seem right, is there meant to be another function that matches this description somewhere?

Or does it remove the line item specified, and _then_ return that line item?

Reply:
It just removed, my mistake. Updated.